Assets/MixedRealityToolkit.Examples/Demos/UX/ManipulationHandler/Scripts/ChangeManipulation.cs

```diff
@@ -31,14 +32,28 @@ public void TryStopManipulation()
         {
             if (manipulatedObject != null && collisionTrigger != null && manipulatedObjCollider != null)
             {
-                if (collisionTrigger.bounds.Intersects(manipulatedObjCollider.bounds))
+                if (!collisionTrigger.bounds.Intersects(manipulatedObjCollider.bounds))
                 {
-                    manipulatedObject.ForceEndManipulation();
+                    return;
+                }
+
+                var mh = manipulatedObject.GetComponent<ManipulationHandler>();
+                var pm = manipulatedObject.GetComponent<PrimitiveManipulator>();
+                if (mh != null)
+                {
+                    mh.ForceEndManipulation();
 
                     // move the object slightly away from the collision point so we can manipulate it again after this
                     Vector3 direction = collisionTrigger.bounds.center - manipulatedObjCollider.bounds.center;
                     manipulatedObject.transform.Translate(direction.normalized * 0.01f);
```

Comment:
If you think that there's any chance that we might add/change these three lines of code, consider changing the structure here to be something more like:

```csharp
if (mh != null || pm != null)
{
  if (mh != null)
  {
     mh.ForceEndManipulation()
  }
  else
  {
     pm.ForceEndManipulation()
  }

  // move the object slightly away from the collision point so we can manipulate it again after this
  Vector3 direction = collisionTrigger.bounds.center - manipulatedObjCollider.bounds.center;
  manipulatedObject.transform.Translate(direction.normalized * 0.01f);
}
```

If not then just resolve this conversation

Reply:
The reason being is that this type of copy paste has the tendency to introduce bugs where someone who is less experienced in the area will fix the "old" version of it (or "new" version) but not update the other, so you find out later and then the next person to fix it will be like "which one is actually correct?"

---
In reply to: [324907573](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5955#discussion_r324907573) [](ancestors = 324907573)